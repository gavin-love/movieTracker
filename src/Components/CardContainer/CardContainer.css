.card_section {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  width: 100%;
  padding: 2%;
}
