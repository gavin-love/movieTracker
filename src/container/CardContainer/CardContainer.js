import React from "react";
import { connect } from "react-redux";
import { Card } from "../../Components/Card/Card";

export const CardContainer = ({ movies }) => {
  const displayMovies = movies.map(movie => {
    console.log(movie);
    return <Card {...movie} key={movie.id} />;
  });
  return <div>{displayMovies}</div>;
};

export const mapStateToProps = state => ({
  movies: state.movies
});

export default connect(
  mapStateToProps,
  null
)(CardContainer);
