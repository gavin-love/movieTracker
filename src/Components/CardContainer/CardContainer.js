import React from "react";
import { connect } from "react-redux";
import { Card } from "../../container/Card/Card";
import "./CardContainer.css";

export const CardContainer = ({ movies }) => {
  const displayMovies = movies.map(movie => {
    return <Card {...movie} key={movie.id} />;
  });
  return <div className="card_section">{displayMovies}</div>;
};

export const mapStateToProps = state => ({
  movies: state.movies
});

export default connect(
  mapStateToProps,
  null
)(CardContainer);
