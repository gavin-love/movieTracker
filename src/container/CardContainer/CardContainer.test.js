import React from "react";
import { CardContainer, mapStateToProps } from "../CardContainer/CardContainer";
import { shallow } from "enzyme";

describe("CardContainer", () => {
  it("should match the snapshot", () => {
    const mockProps = {
      movies: [{ id: 1 }, { id: 2 }, { id: 3 }]
    };

    const wrapper = shallow(<CardContainer {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  describe("mapStateToProps", () => {
    const mockState = {
      movies: [{}, {}, {}]
    };

    const expected = {
      movies: [{}, {}, {}]
    };

    const mappedProps = mapStateToProps(mockState);

    expect(mappedProps).toEqual(expected);
  });
});
