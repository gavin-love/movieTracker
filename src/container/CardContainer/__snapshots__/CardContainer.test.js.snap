// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    movies={
        Array [
            Object {
              "id": 1,
            },
            Object {
              "id": 2,
            },
            Object {
              "id": 3,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          id={1}
/>,
        <Card
          id={2}
/>,
        <Card
          id={3}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "id": 1,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "id": 2,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "function",
        "props": Object {
          "id": 3,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            id={1}
/>,
          <Card
            id={2}
/>,
          <Card
            id={3}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "id": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "id": 2,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "id": 3,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
